test/conftest.py

```diff
@@ -85,23 +91,51 @@ def patch_logging(moulinette):
         },
     }
 
-    moulinette.init(
-        logging_config=logging,
-        _from_source=False
-    )
-
 
 @pytest.fixture(scope='session', autouse=True)
-def moulinette():
+def moulinette(tmp_path_factory):
     import moulinette
 
+    # Can't call the namespace just 'test' because
+    # that would lead to some "import test" not importing the right stuff
+    namespace = "moulitest"
+    tmp_cache = str(tmp_path_factory.mktemp("cache"))
+    tmp_data = str(tmp_path_factory.mktemp("data"))
+    tmp_lib = str(tmp_path_factory.mktemp("lib"))
+    os.environ['MOULINETTE_CACHE_DIR'] = tmp_cache
+    os.environ['MOULINETTE_DATA_DIR'] = tmp_data
+    os.environ['MOULINETTE_LIB_DIR'] = tmp_lib
+    shutil.copytree("./test/actionsmap", "%s/actionsmap" % tmp_data)
+    shutil.copytree("./test/src", "%s/%s" % (tmp_lib, namespace))
+    shutil.copytree("./test/locales", "%s/%s/locales" % (tmp_lib, namespace))
+
     patch_init(moulinette)
     patch_translate(moulinette)
-    patch_logging(moulinette)
+    logging = patch_logging(moulinette)
+
+    moulinette.init(
+        logging_config=logging,
+        _from_source=False
+    )
 
     return moulinette
 
 
+@pytest.fixture(scope='session')
+def moulinette_webapi(moulinette):
+
+    namespace = "moulitest"
+
+    api_thread = Process(target=moulinette.api,
+                         args=([namespace],),
+                         kwargs={"host": "localhost", "port": 12342, "use_websocket": False})
+    api_thread.start()
+    time.sleep(0.5)
```

Comment:
`sleep` in tests is fine at first but can build up (as others see it is "already being used"). Perhaps a while loop with a counter to break out? Things start faster on some machines, slower on others? Alternatively just leave it (so much to be done now anyway) and wait to hear if it causes issues.

Reply:
Got reworked in https://github.com/YunoHost/moulinette/pull/216/commits/53ab4709d167ca70b4288a36706de33b1bb60221